src/configure/helper/templateHelper.ts

```diff
@@ -176,45 +178,52 @@ export async function getFilteredTemplates(azureSession: AzureSession, resourceT
+                filteredTemplates = await client.getTemplatesInfoByFilter("docker", "Azure:AKS");
+            }, TelemetryKeys.TemplateServiceDuration);
             return filteredTemplates;
         default:
             return null;
     }
 }
 
+export async function getTemplates(azureSession: AzureSession, repoAnalysisParameters: RepositoryAnalysis, githubPatToken?: string) {
+    const client = await TemplateServiceClientFactory.getClient(azureSession.credentials, githubPatToken);
+    let templates: TemplateInfo[];
+    await telemetryHelper.executeFunctionWithTimeTelemetry(async () => {
+        templates = await client.getTemplates(repoAnalysisParameters);
+    }, TelemetryKeys.TemplateServiceDuration);
+    return templates;
+}
+
 export async function analyzeRepoAndListAppropriatePipeline2(azureSession: AzureSession, sourceRepository: GitRepositoryParameters, pipelineType: PipelineType, repoAnalysisParameters: RepositoryAnalysis, githubPatToken?: string, resource?: GenericResource): Promise<PipelineTemplate[]> {
 
     let pipelineTemplates: PipelineTemplate[] = [];
     let remoteTemplates: TemplateInfo[] = [];
     let localPipelineTemplates: LocalPipelineTemplate[] = await this.analyzeRepoAndListAppropriatePipeline(sourceRepository, repoAnalysisParameters);
 
-    if (repoAnalysisParameters && repoAnalysisParameters.applicationSettingsList && pipelineType === PipelineType.GitHubPipeline) {
+    if (pipelineType === PipelineType.GitHubPipeline) {
         try {
-            const client = await TemplateServiceClientFactory.getClient(azureSession.credentials, githubPatToken);
-
-            await telemetryHelper.executeFunctionWithTimeTelemetry(async () => {
-                if (!!resource) {
-                    // If resource is already selected, getting templates accordingly
-                    localPipelineTemplates = [];
-                    remoteTemplates = await getFilteredTemplates(azureSession, resource.type, githubPatToken);
-                } else {
-                    remoteTemplates = await client.getTemplates(repoAnalysisParameters);
-                }
-            }, TelemetryKeys.TemplateServiceDuration);
+            if (!!resource) {
+                localPipelineTemplates = [];
+                remoteTemplates = await getFilteredTemplates(azureSession, resource.type, githubPatToken);
+            }
+            else if (repoAnalysisParameters && repoAnalysisParameters.applicationSettingsList) {
+                remoteTemplates = await getTemplates(azureSession, repoAnalysisParameters, githubPatToken);
+            }
             pipelineTemplates = await convertToPipelineTemplate(remoteTemplates);
+            pipelineTemplates = pipelineTemplates.concat(localPipelineTemplates);
+            // sorted by weight
+            pipelineTemplates = pipelineTemplates.sort((a, b) => {
+                return b.templateWeight - a.templateWeight;
```

Comment:
you have changed the sorting algo. it was returning 1/-1 but now can return any integer. is this intentional and needed? @kanika1894

Reply:
This is just to simplify the code.
It won't change the logic. The functioning of sort function is in such a way that it will just consider whether the difference is positive or negative, and sort accordingly.
More examples : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Reply:
Makes sense